File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/Readonly/TopicListItem.cs
using MaterialAdvisor.Application.Models.Shared;

namespace MaterialAdvisor.Application.Models.Readonly;

public class TopicListItem
{
    public Guid Id { get; set; }

    public string Owner { get; set; }

    public ushort Number { get; set; }

    public IEnumerable<LanguageText> Texts { get; set; } = [];

    public IEnumerable<KnowledgeCheckListItem> KnowledgeChecks { get; set; } = [];
}

File: src/app/material-advisor/src/app/features/main-page/models/TopicListItem.ts
import { LanguageText } from "@shared/models/LanguageText";
import { GUID } from "@shared/types/GUID";
import { KnowledgeCheckListItem } from "./KnowledgeCheckListItem";

export interface TopicListItem {
  id: GUID;
  number: number;
  owner: string;
  name: LanguageText[];
  knowledgeChecks: KnowledgeCheckListItem[];
}
